src/hotspot/share/oops/method.cpp

```diff
@@ -507,12 +508,33 @@ void Method::print_invocation_count() {
   }
   tty->cr();
 
-  tty->print_cr ("  interpreter_invocation_count: %8d ", interpreter_invocation_count());
-  tty->print_cr ("  invocation_counter:           %8d ", invocation_count());
-  tty->print_cr ("  backedge_counter:             %8d ", backedge_count());
+  // Internal counting is based on signed int counters. They tend to
+  // overflow with longer-running workloads on fast machines. To shift
+  // the overflow limit, we interpret the return value as unsigned int.
+  // This is ok because counters are unsigned by nature, and it gives us
+  // another factor of 2 before the counter values become meaningless.
+  // Print a "overflow" notification to create awareness.
+  const char* addMsg;
+  unsigned int maxInt = (1U<<31) - 1;
```

Comment:
Why not use INT_MAX?

Reply:
Will change.

Reply:
With overflow I do not mean "counter overflow" as it is used to trigger compiler activities but plain simple "range of singed int exceeded". Should I use different wording? E.g. "counter in signed int overflow"?

Reply:
Then it will never happen. These values come from ```InvocationCounter::count()```. And it will never return a value > 2^31 - 1.

Reply:
Sorry, it can return 2^31. Which would be an indication of a counter overflow (InvocationCounter::count_limit == 2^31). Your code is correct in both senses. It is a counter overflow and a signed int overflow.

Reply:
Oops, my arithmetic is bad again. InvocationCounter::count_limit is 2^30. So, I don't think there ever going to be an overflow that you're looking for. The sign bit is always 0. Or am I missing something again? :)